bundles/org.openhab.binding.modbus.sungrow/src/main/java/org/openhab/binding/modbus/sungrow/internal/SungrowInverterHandler.java

```diff
@@ -157,6 +170,60 @@ public void modbusInitialize() {
         }
     }
 
+    private void updateProperties(SungrowInverterConfiguration config) {
+        ModbusReadRequestBlueprint getVersionRequest = new ModbusReadRequestBlueprint(this.getSlaveId(),
+                ModbusReadFunctionCode.READ_INPUT_REGISTERS, //
+                4949, //
+                33, //
+                config.maxTries //
+        );
+        this.submitOneTimePoll(getVersionRequest, this::updateVersionProperties, this::readError);
+
+        ModbusReadRequestBlueprint getDeviceInfo = new ModbusReadRequestBlueprint(this.getSlaveId(),
+                ModbusReadFunctionCode.READ_INPUT_REGISTERS, //
+                4989, //
+                13, //
+                config.maxTries //
+        );
+        this.submitOneTimePoll(getDeviceInfo, this::updateDeviceInfoProperties, this::readError);
+    }
+
+    private void updateVersionProperties(AsyncModbusReadResult result) {
+        result.getRegisters().ifPresent(registers -> {
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+            long protocolNo = ModbusBitUtilities.extractUInt32(registers.getBytes(), 0);
+            getThing().setProperty("Protocol No", String.valueOf(protocolNo));
+            long protocolVersion = ModbusBitUtilities.extractUInt32(registers.getBytes(), 4);
+            getThing().setProperty("Protocol Version", String.valueOf(protocolVersion));
+            String certVersionArm = ModbusBitUtilities.extractStringFromRegisters(registers, 4, 28,
+                    StandardCharsets.UTF_8);
+            getThing().setProperty("Certification version number of ARM software", certVersionArm);
+            String certVersionDsp = ModbusBitUtilities.extractStringFromRegisters(registers, 19, 28,
+                    StandardCharsets.UTF_8);
+            getThing().setProperty("Certification version number of DSP software", certVersionDsp);
```

Comment:
Please extract property keys to a constant (preferably in the bindingconstants file) and they should also be camelcase:
For reference: https://www.openhab.org/docs/developer/guidelines.html#naming-convention